lib/auth/auth_with_roles.go

```diff
@@ -56,6 +56,20 @@ func (a *AuthWithRoles) currentUserAction(username string) error {
 		defaults.Namespace, services.KindUser, services.VerbCreate)
 }
 
+// authConnectorAction is a special checker that grants access to auth
+// connectors. It first checks if you have access to the specific connector.
+// If not, it checks if the requester has the meta KindAuthConnector access
+// (which grants access to all connectors).
+func (a *AuthWithRoles) authConnectorAction(namespace string, resource string, verb string) error {
+	if err := a.checker.CheckAccessToRule(&services.Context{User: a.user}, namespace, resource, verb); err != nil {
+		if err := a.checker.CheckAccessToRule(&services.Context{User: a.user}, namespace, services.KindAuthConnector, verb); err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
+	return nil
+}
+
```

Comment:
I think this function has a logic error. If the 1st check fails but the 2nd succeeds, it will succeed. It's basically an "OR" instead of "AND".

Reply:
That's intended it's supposed to be OR. It's a fully backward compatible change.